<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1014</width>
    <height>698</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Widget</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="1">
    <widget class="QPushButton" name="show">
     <property name="text">
      <string>显示</string>
     </property>
    </widget>
   </item>
   <item row="0" column="2">
    <widget class="QPushButton" name="showTable">
     <property name="text">
      <string>showTable</string>
     </property>
    </widget>
   </item>
   <item row="1" column="1" rowspan="2">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string/>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QPushButton" name="attack">
        <property name="text">
         <string>攻击</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="add">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>添加</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="detect">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>检测</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="backup">
        <property name="text">
         <string>备份</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="restore">
        <property name="text">
         <string>恢复</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="2">
    <widget class="QTableView" name="tableView"/>
   </item>
   <item row="2" column="0">
    <widget class="QProgressBar" name="progressBar">
     <property name="value">
      <number>0</number>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>select_connect</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QPushButton" name="set_database">
            <property name="text">
             <string>connect</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Database</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="database"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Table</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="table"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Key</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEdit">
            <property name="placeholderText">
             <string>input you key</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="select_key">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>参数设置</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_10">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>水印比例系数</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="scale">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>999</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>最不重要比特数</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="lsb">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>64</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>阈值</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="threshold"/>
            </item>
            <item>
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>percent</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="percent"/>
            </item>
            <item>
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>error_rate</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="error_rate">
              <property name="decimals">
               <number>5</number>
              </property>
              <property name="singleStep">
               <double>0.000010000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_9">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>可以添加的属性</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTextEdit" name="can_add">
              <property name="readOnly">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QPushButton" name="to_right">
              <property name="text">
               <string>-----&gt;</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="to_left">
              <property name="text">
               <string>&lt;-----</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>不添加水印的属性</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTextEdit" name="can_not_add">
              <property name="readOnly">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
